ui-tests/src/test/java/org/openmrs/reference/page/AppointmentBlocksPage.java

```diff
@@ -131,6 +131,7 @@ public void clickOnleft() {
      */
     public void clickOnDay() {
         clickOn(DAY);
+        waiter.until(ExpectedConditions.elementToBeClickable(DAY_CLASS));
```

Comment:
Often the `click() `on  `DAY_CLASS` is intercepted by the element above  it and this results into `ElementClickInterceptedException`. Though `JavascriptExecutor` from selenium `API`  can retrieve elements that are obscured by others But Invoking `elementToBeClickable() ` is precise and paves the way.